README.md

```diff
@@ -1,9 +1,284 @@
+    ```json
+    {
+      "comment": {
+        "username": "butter_bridge",
+        "body": "Massive intercranial brain haemorrhage",
+        "votes": 0,
+        "author": "icellusedkars",
+        "article_id": 1,
+        "created_at": 1583133000000
+      }
+    }
+    ```
+
+- `PATCH /api/articles/:article_id`
+
+  - Description: Increase or decrease votes for an article
+  - Request Body:
+    Increase votes
+    ```json
+    {
+      "inc_votes": 5
+    }
+    ```
+    Decrease votes
+
+  ```json
+  {
+    "inc_votes": -5
+  }
+  ```
+
+  - Example Response:
+    ```json
+    {
+      "updatedArticle": {
+        "title": "Seafood substitutions are increasing",
+        "topic": "cooking",
+        "author": "weegembump",
+        "created_at": "2018-05-30T15:59:13.341Z",
+        "votes": 5,
+        "comment_count": 6
+      }
+    }
+    ```
+
+- `DELETE /api/comments/:comment_id`
+
+  - Description: Delete a comment by comment_id
+  - Status Codes:
+    - `204`: No content (successful deletion)
+    - `404`: Not Found (comment not found)
+
+- `GET /api/users`
+  - Description: Get all users
+  - Example Response:
+    ```json
+    {
+      "users": [
+        {
+          "username": "butter_bridge",
+          "name": "Maurice Moss",
+          "avatar_url": "https://example.com/avatar/butter_bridge.jpg"
+        }
+      ]
+    }
+    ```
+
+## Concepts
+
+The API encompasses several concepts, including:
+
+- RESTful API design
+- Database schema design
+- SQL querying and manipulation
+- CRUD operations (Create, Read, Update, Delete)
+- Data modeling, controllers and relationships (based on MVC pattern)
+- Error handling, middleware and status codes (200, 201, 204, 400, 404, 500)
+
+## Technologies
+
+The API is built using the following technologies:
+
+- Express
+- Node.js
+- PostgreSQL
+- Git & GitHub
+
+## Features
+
+The API offers the following features:
+
+- Retrieve a list of available endpoints
+- Get topics and articles with various filtering options
+- Retrieve comments for a specific article
+- Add comments to articles
+- Update votes on articles
+- Delete comments
+- Retrieve user information
+
+## Prerequisites
+
+Before you begin, make sure you have the following software installed on your machine:
+
+- Node.js (minimum version: 14.x)
+- PostgreSQL (minimum version: 13.x)
+
+## Clone the Repository
+
+```sh
+git clone https://github.com/jcainuk/be-nc-news
+cd be-nc-news
+
+```
+
+## Install dependencies
+
+```
+npm install
+
+```
+
+## Create .env Files
+
+Create two `.env` files for development and testing environments:
+
+1. Create a `.env.development` file in the project root and add the following:
+
+```env
+PGDATABASE=your_dev_database_name
+
+
+```
+
+2. Create a `.env.test` file in the project root and add the following:
+
+```env
+PGDATABASE=your_test_database_name
+
+```
+
+Please make sure to replace `your_dev_database_name` and `your_test_database_name` with your actual database and user credentials.
```

Comment:
This bit needs correcting! The setup.sql file has the names of the databases - it won't work with just any database names. You should either put them in the example, or point the user towards setup.sql.

For reference, the contents of it are:

```
DROP DATABASE IF EXISTS nc_news_test;
DROP DATABASE IF EXISTS nc_news;

CREATE DATABASE nc_news_test;
CREATE DATABASE nc_news;
```

Reply:
Thank you so much :)